benches/Cargo.toml

```diff
@@ -0,0 +1,21 @@
+[package]
+name = "benches"
+version = "0.0.0"
+publish = false
+edition = "2018"
+
+[dependencies]
+linkerd2-cache = { path = "../linkerd/cache" }
+linkerd2-error = { path = "../linkerd/error" }
+linkerd2-lock = { path = "../linkerd/lock" }
+linkerd2-stack = { path = "../linkerd/stack" }
+bencher = "0.1.5"
```

Comment:
Elsewhere, we have benchmarks that are written with the libtest benchmark framework. I'm not opposed to switching to `bencher`, especially since it can run on stable; however, I think it would be good to pick one and use it consistently. If we decide to adopt `bencher` here, we should probably open a follow-up to change other benchmarks to use `bencher` as well.

We may also want to consider `criterion` rather than `bencher`. Although I have my doubts about the accuracy of `criterion`'s regression detection (it seems way too sensitive to noise IMO), I think some of the additional benchmarking APIs `criterion` provides, like [`bench_with_input`](https://docs.rs/criterion/0.3.1/criterion/struct.BenchmarkGroup.html#method.bench_with_input) and [`iter_custom`](https://docs.rs/criterion/0.3.1/criterion/struct.Bencher.html#method.iter_custom) are very useful when writing benchmarks...just a thought!